test: Turn BIP44 on by default

davibroc opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

Until now, only the tests in the multichain-accounts folder have run with BIP44 enabled. This PR enables BIP44 by default for all tests. We achieve this by mocking the BIP44 feature [flag here.](https://github.com/MetaMask/metamask-extension/blob/3b17ad4e36b2ca8d2ae460981c9069cbf11a1b97/test/e2e/helpers.js#L336-L339) Once this PR is merged and the feature flag is turned on, this mock will be removed.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/37004?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.









<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Enables BIP44 stage 2 across E2E by mocking the feature flag and updates page objects, helpers, mocks, and tests (balances, selectors, flows, hardware/Solana/bridge) to align with the new multichain UI and behavior.
> 
> - **Test infra & flags**:
>   - Enable BIP44 stage 2 by default via feature-flag mocks; default `forceBip44Version` on; propagate `BIP44_STAGE_TWO` to mocks (bridge, confirmations, BTC/Solana).
>   - Add price API mocks; refine helpers (login waits, websocket server usage) and new hardware wallets helper.
> - **Page Objects**:
>   - Update selectors and flows (account list, header navbar, send page, asset list, activity list, multichain account/details, address list, edit-connected-accounts modal) to new multichain UI; add methods for multichain wallet/account ops; adjust balance privacy and copy-address behavior.
> - **Fixtures/Builders**:
>   - Add preference helpers (show native token as main balance); tweak TokensController test data; expand enabled networks.
> - **Tests**:
>   - Widespread updates for BIP44 behavior: balances (fiat/native), network expectations, activity selectors (`.transaction-status-label--confirmed`), account names, and flows.
>   - Adjust/skip cases for current BIP44 limitations; update hardware (Ledger/Trezor/QR) and Solana flows; bridge/swap tests use updated mocks; metrics use stable IDs.
>   - Add multichain websocket tests; refactor multi-SRP tests and permissions flows.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e36b02788aeb7e45638fa08463b958eb18c9fa83. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

davibroc committed:
Attempt to add Solanaa account

davibroc committed:
Cursor fix

davibroc committed:
Changed default fixture

davibroc committed:
Added debug test script

davibroc committed:
Fix lint

seaona reviewed (dismissed):
we shouldn't update the default fixture as this wallet state is mixed between old and new.

For updating the fixture to the latest state, there's this whole Epic planned here, which is currently in progress:

https://consensyssoftware.atlassian.net/browse/MMQA-443

davibroc committed:
Fixed keyring

davibroc committed:
Add price API mock

davibroc committed:
Fix lint

davibroc committed:
test: Fixed some tests

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

This PR turns BIP44 by default and make changes to the fixtures so that
Solana account is already created when the Extension starts

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/37365?quickstart=1)

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

Fixes:

1. Go to this page...
2.
3.

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

<!-- [screenshots/recordings] -->

<!-- [screenshots/recordings] -->

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Enables BIP-44 state 2 by default across e2e, adds multichain/Solana
defaults to fixtures, and updates page objects and tests to the new
multichain flows and feature-flag mocks.
>
> - **E2E Framework**:
> - Default `withFixtures` now enables BIP-44 stage 2 via feature flag
(boolean `forceBip44Version`, true by default) and applies only state-2
mocks when enabled.
> - Updated selectors and flows in benchmarks (use
`HeaderNavbar`/`AccountListPage`; activity selector for confirmed tx).
> - **Fixtures & Mocks**:
> - `default-fixture` seeds `AccountTreeController` with a multichain
wallet (Ethereum + Solana account) and updates `AccountsController`
metadata; enables Solana in network enablement.
> - Introduces `BIP44_STAGE_TWO` and injects it into feature-flag
responses; various bridge fixtures/mocks include it.
> - **Page Objects**:
> - `AccountListPage.checkPageIsLoaded` simplified to multichain UI
(always expects "Create account" and options menu; removes state arg);
ensures "Syncing" not shown.
>   - `HeaderNavbar` adds `clickNetworkAddresses`.
> - **Bridge Tests**:
> - Adjusted expectations to ETH balances/units; explicitly disable
BIP-44 where needed; minor skips added; hardforks normalized; duplicate
mocks removed.
> - **Multichain Accounts Tests**:
> - Updated to new `checkPageIsLoaded()` signature and multichain flows;
remove per-test state flags; minor assertions streamlined.
> - Unskips and simplifies wallet details scenario (validate presence in
account list).
> - **Solana Test Utilities**:
> - Add native price mock; tweak enabled networks and initial navigation
(network filter steps); remove per-account creation loop.
> - **Misc**:
> - Numerous tests now pass `forceBip44Version: false` where legacy
behavior is required; balance constants adjusted.
>
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
f8bb58746570dc0d8221589467490686b98cf129. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

davibroc committed:
Fixed send test

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/37406?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Hardcodes local-node balance to 24.998 in homepage tests and enables
the sendRedesign feature flag in multichain accounts mocks.
> 
> - **E2E Tests**:
> - **Homepage balance check**
(`test/e2e/page-objects/pages/home/homepage.ts`):
> - For local nodes, replace dynamic balance fetch with fixed `"24.998"`
in `checkLocalNodeBalanceIsDisplayed`.
> - **Feature flags**
(`test/e2e/tests/multichain-accounts/feature-flag-mocks.ts`):
>     - Set `sendRedesign.enabled` to `true` in `BIP44_STAGE_TWO`.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
aea2615f2e70ca2b795d80df7eb3868085936d98. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

davibroc committed:
Fix more tests

davibroc committed:
Fixed more tests

davibroc committed:
Fix benchmark tests

davibroc committed:
Fixed hardware wallets

davibroc committed:
Fixed transactions

davibroc committed:
Fix lint

davibroc committed:
Various fixes

davibroc committed:
Fixed onboarding

davibroc committed:
Fixed confirmation part 1

davibroc committed:
Various fixes

davibroc committed:
Fix tokens tests

davibroc committed:
Fix fixture builder

davibroc committed:
Fixed metric tests

davibroc committed:
Fix account test

davibroc committed:
Fix snaps

davibroc committed:
Fix more tests

davibroc committed:
Fix more tests

davibroc committed:
Fix snap tests

davibroc committed:
Fix more tests

davibroc committed:
Fix BTC

davibroc committed:
Fix flask

davibroc committed:
Fix multisrp flask

davibroc committed:
Fix flask tests

davibroc committed:
Fix bug

davibroc committed:
Fix lint

davibroc committed:
Fix all

davibroc committed:
Fix send

davibroc committed:
Fixed account settings

jiexi commented on test/e2e/flask/multichain-api/non-evm/wallet_createSession.spec.ts:
```diff
@@ -93,8 +91,7 @@ describe('Multichain API - Non EVM', function () {
 
-          // 0 index is select all, 1 index is EVM default account, 2 index is Solana account 1 (default)
-          const accountCheckbox = checkboxes[2];
+          const accountCheckbox = checkboxes[0];
```
should this be `checkboxes[1]` instead?

davibroc replied:
@jiexi 
Actually the test description is  `call `wallet_createSession` with Solana scope without any accounts requested should automatically select the current active Solana account`. Before we had:
- Select All
- Account 1
- Solana Account 1
- Account 2

But now `Select All` and `Solana Account 1` are gone. So the automatically selected one is the first one `Account 1` as you see below, which includes a Solana address. Does it make sense?
<img width="1193" height="820" alt="Screenshot 2025-11-06 at 6 00 48 PM" src="https://github.com/user-attachments/assets/a63ed883-42ea-4d4e-ab2b-54a29ab03c16" />

jiexi commented on test/e2e/flask/solana-wallet-standard/signMessage.spec.ts:
```diff
@@ -9,7 +9,9 @@ import {
-describe('Solana Wallet Standard - Sign Message', function () {
+// Skipping this test because it's failing with BIP44
+// eslint-disable-next-line mocha/no-skipped-tests
+describe.skip('Solana Wallet Standard - Sign Message', function () {
```
same question

jiexi commented on test/e2e/flask/multichain-api/non-evm/wallet_invokeMethod.spec.ts:
```diff
@@ -7,7 +7,9 @@ import { withSolanaAccountSnap } from '../../../tests/solana/common-solana';
-describe('Multichain API - Non EVM', function () {
+// Skipping these tests because it's failing with BIP44
+// eslint-disable-next-line mocha/no-skipped-tests
+describe.skip('Multichain API - Non EVM', function () {
```
nevermind. I read the PR desc

davibroc committed:
removed onboard and selectAll option from connectSolanaTestDapp

davibroc commented on test/e2e/flask/solana-wallet-standard/connect.spec.ts:
```diff
@@ -31,7 +31,7 @@ describe('Solana Wallet Standard - e2e tests', function () {
           await testDapp.checkPageIsLoaded();
 
-          await connectSolanaTestDapp(driver, testDapp, { onboard: true });
+          await connectSolanaTestDapp(driver, testDapp, { onboard: false });
```
Good point. I have removed the `onBoard` and `selectAllAccounts` option and related code

davibroc committed:
Fix bridge

davibroc committed:
Fix FF

davibroc committed:
Fixed more tests

davibroc committed:
Fix flask

davibroc committed:
Fix failures

davibroc committed:
Fix flask

gantunesr commented on test/e2e/tests/account/unlock-wallet.spec.ts:
```diff
@@ -96,6 +98,8 @@ describe('Unlock wallet - ', function () {
         await loginPage.loginToHomepage(WALLET_PASSWORD);
 
+        // Wait syncing to finish
+        await driver.delay(10_000);
```
syncing shouldn't take this long? @mathieuartu wdyt?

davibroc commented on test/e2e/flask/btc/btc-account-overview.spec.ts:
Done

davibroc committed:
Fix flask and address comments

davibroc committed:
Removed fixture changes

davibroc committed:
Fix tests given UI changes after rebase

davibroc commented on test/e2e/tests/multichain/aggregated-balances.spec.ts:
```diff
@@ -20,7 +20,9 @@ const NETWORK_NAME_MAINNET = 'Ethereum';
-describe('Multichain Aggregated Balances', function (this: Suite) {
+// This test fails on BIP44
+// eslint-disable-next-line mocha/no-skipped-tests
+describe.skip('Multichain Aggregated Balances', function (this: Suite) {
```
By the way we already have similar test to check the balance here `test/e2e/tests/multichain-accounts/account-details.spec.ts`

davibroc committed:
Add information about BIP44 failures

davibroc committed:
Add more info

davibroc committed:
Added more bug details and fixed some tests

davibroc committed:
Fixed multichain evm test

davibroc committed:
Changed skip

davibroc committed:
Fix dapp send transaction

davibroc committed:
Removed skip to some of the tests

davibroc committed:
Fixed account balance

davibroc committed:
Fix balance tests

davibroc committed:
Fix solana

davibroc committed:
Fix snap test

seaona commented on test/e2e/page-objects/pages/home/activity-list.ts:
```diff
@@ -55,8 +55,8 @@ class ActivityListPage {
-  private readonly pendingTransactionItems =
-    '.transaction-list__pending-transactions .activity-list-item';
+  private readonly unapprovedTransactionItems =
+    '.transaction-status-label--unapproved';
```
Unapproved transactions are not the same as Pending transactions. This should be revised

davibroc replied:
The pendingTransactionItems was not used by any methods so I removed it. I needed `unapprovedTransactionItems` action which is used by method `checkUnapprovedTxNumberDisplayedInActivity`

seaona commented on test/e2e/page-objects/pages/multichain/multichain-account-details-page.ts:
```diff
@@ -165,6 +164,15 @@ class MultichainAccountDetailsPage {
+  async clicRevealRow(): Promise<void> {
+    console.log('Click on the private key row');
+    const revealSrpRow = await this.driver.findElement(this.exportSrpButton);
+    await revealSrpRow.click();
```
we should avoid finding an element and then using .click. Instead we should use our driver method: clickElement, which has guards and prevents race conditions like StaleElement errors

davibroc replied:
Done

seaona commented on test/e2e/page-objects/pages/test-dapp-multichain.ts:
```diff
@@ -202,6 +202,7 @@ class TestDappMultichain {
   }> {
     await this.driver.switchToWindowWithTitle(WINDOW_TITLES.MultichainTestDApp);
     await this.clickWalletGetSessionButton();
+    await this.driver.delay(1000);
```
is there a condition we can wait here instead of the delay?

davibroc replied:
Removed

seaona commented on test/e2e/tests/account/forgot-password.spec.ts:
```diff
@@ -33,6 +33,7 @@ describe('Forgot password', function () {
         localNodes: Anvil[] | Ganache[] | undefined[];
       }) => {
         await loginWithBalanceValidation(driver, localNodes[0]);
+        await driver.delay(5000);
```
same, is there a reason for this long delay? Could this be changed by a condition? if not, we might want to add a comment explaining why this is needed

davibroc replied:
I reduced this to 3 secs and added comment. We need to give sometime for network calls to settle before locking metamask otherwise the test will fail

davibroc committed:
Fix websocket

davibroc committed:
Fix after rebase

davibroc committed:
Fix advance functionality

davibroc committed:
Fix seedless onboarding

salimtb committed:
fix: fix e2e tests

davibroc committed:
Fix tests after salim PR

davibroc committed:
Fix 3 failures

davibroc committed:
Fix more tests

davibroc committed:
Fix state logs

davibroc commented on test/e2e/tests/vault-corruption/vault-corruption.spec.ts:
```diff
@@ -255,12 +259,22 @@ describe('Vault Corruption', function () {
+
+    const addressListModal = new AddressListModal(driver);
+    await addressListModal.clickQRbutton();
+    await driver.delay(1000);
```
Removed

davibroc commented on test/e2e/tests/vault-corruption/vault-corruption.spec.ts:
```diff
@@ -8,10 +8,13 @@ import {
-describe('Vault Corruption', function () {
+// Skipping this test as not working with BIP44
+// eslint-disable-next-line mocha/no-skipped-tests
+describe.skip('Vault Corruption', function () {
```
I added BUG #38080 - Reactivate vault corruption test after BIP44 is turned on with FF

davibroc committed:
Fixes based on review comments

davibroc committed:
Fix solana tests

davibroc committed:
Fix test failing after Salim changes

davibroc committed:
Fix failing test

davibroc committed:
Clean up

davibroc committed:
Fixed more tests

davibroc committed:
Try different fix

davibroc committed:
test: Fix test failures after rebase (#38154)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/38154?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Stabilizes and refactors multichain and snap account E2E tests by
standardizing login flows, simplifying setup, adjusting
delays/mocks/fixtures, and reworking the hardware wallet test to use
direct fixtures with local balance setup.
> 
> - **E2E: Snap account transfers
(`test/e2e/tests/account/snap-account-transfers.spec.ts`)**
> - Standardize on `loginWithBalanceValidation`; remove
`loginWithoutBalanceValidation` usage.
> - Adjust timing: add pre-send delay for sync flow; keep post-send
delay for async approve.
> - Update fixtures (enable networks/prefs) and mocks wiring; add
`ignoredConsoleErrors` for reject case.
> - Update balance assertions; comment out flaky post-tx balance check
in async approve.
> - **E2E helper refactor
(`test/e2e/tests/multichain-accounts/common.ts`)**
> - Remove `NetworkManager` navigation; open account menu directly after
login.
> - Split fixtures by `AccountType` (Ledger vs MultiSRP) with
prefs/network enabled; tweak mock aggregation.
> - Use `loginWithoutBalanceValidation` only for hardware wallets;
others use `loginWithBalanceValidation`.
> - **E2E: Multichain account list menu
(`test/e2e/tests/multichain-accounts/multichain-account-list-menu.spec.ts`)**
> - Rework hardware wallet test to use `withFixtures`, set local node
balance, and log in with balance validation; add necessary
mocks/imports.
> - Minor cleanup and consistency updates across tests
(headers/navigation, balances).
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
d5069223cb4ceef46ddfdbe3a12cd7fc5880adbc. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

gantunesr reviewed (dismissed):
Looks good for Accounts tests. Left a few comments but those can be addressed on a follow up PR

mikesposito commented on test/e2e/fixture-builder.js:
```diff
@@ -169,7 +169,7 @@ class FixtureBuilder {
     return this.withKeyringController({
       vault:
-        '{"data":"tM9QywcUa46iRvWsfvOL9mJqOrRLoVZoDuqTwxEt1Jz4qCpVIx8I/+7wMQuHBhe+DLBMSB7DzWeBpkCngTSE/mt6ygXWd96aKPH00PCW7uq/Z+8gdHQ3+ZGVCkTIDvLwOzG2gywrOfWRzRRcFwV545EV2iC6Q47A6KcgK/YokBeT4uVJ+oC309490eYn6/LkC+e+DNzJOlESs0LOynMJPMP0Wc53AvEuVlmYA2QLUKa+X6Eo1FEm91lg7znnNGTH7d7PVzDjElTQAUcQmiCvLfJU3cCmnVubarG/eOPWyL41u1z2IFMuf2QKoJNG7garFS+z4THtqWuR/NiYbNCJ70G6V2P0+9ntIWMk4qs4cBY4Pl3MPsyXBVhVoL+sL ...
+        '{"data":"1bMxVc4PgogxGePtaCqfWAm3/ASfbOwnw6kGskZFqNoc/vSECsDK7eeWHZpybG2lG29KBQzMfDyy0Dt1kNcsU8DyP6LdJtSzOwm4TLu9QoSUSM8X365ae2zShi2MJfKHvqCRwhx4Dcx90Gz+Ddx2pmLjmMU2RWFC50cnKo+p1Ml3f7XaC+BqLRrpGipKolY7GJDCA/NFvW6JAYWsBOrQKClswb2xAh5AcZRIVT4m31XVEfJNT8z1Z+6J8xZu1vSJd6UEl6PWF53gu4aBAYvSpshXHqMG76WbAGvwS+7l53pC1VGN5TNY3bu+f6H03HXBMXkhfscVf1LljNSiKuHPsZoCNCegtiqEjQwqpGLDKHyujq0NsL30lBk7B1ucl ...
```
There are two different `SnapKeyring` in this vault, it should never be possible. Not sure if it is causing any issue, but probably best to remove the second one so we avoid subtle issues in the future

```
[
  ...,
  {
    "type": "Snap Keyring",
    "data": {
      ...
    },
    "metadata": {
      "id": "01K90ME8BAWA3BCEHPDHB0SGG7",
      "name": ""
    }
  },
  {
    "type": "Snap Keyring",
    "data": {
      "accounts": {}
    },
    "metadata": {
      "id": "01K90ME8BB6N91MZ5ARVPSEM3J",
      "name": ""
    }
  },
  {
    "type": "HD Key Tree",
    "data": {
      ...
    },
    "metadata": {
      "id": "01K90MKV5XAH4Q3F75MFHKBX75",
      "name": ""
    }
  }
]

davibroc committed:
test: Remove modified vaults (#38207)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/38207?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Replace modified vault fixtures and update E2E tests (snaps labels,
network counts, metrics), refactor websocket tests, and revise
state-logs schema and spec.
> 
> - **Fixtures**:
> - Replace encrypted `vault` payloads in `test/e2e/default-fixture.js`
and `fixture-builder.js` (`withKeyringControllerAdditionalAccountVault`,
`withKeyringControllerMultiSRP`, default vault).
> - **Snaps E2E**:
> - Update entropy source labels from long IDs to `SRP 1 (primary)` /
`SRP 2` in `test-snap-bip-32.spec.ts`, `test-snap-bip-44.spec.ts`, and
`test-snap-get-entropy.spec.ts`.
> - **Connections/Permissions E2E**:
> - Adjust expected connected networks count from `12` to `11` in
`edit-networks-permissions.spec.ts` and
`multiple-provider-connections.spec.ts`.
> - **Metrics E2E**:
> - Update `DappViewed` assertions for `number_of_accounts` from `3` to
`1` in `dapp-viewed.spec.ts`.
> - **WebSocket E2E**:
>   - Remove `tests/multichain-accounts/web-socket-connection.spec.ts`.
> - Refactor `tests/multichain/web-socket-connection.spec.ts` to use
`waitForWebsocketConnections` helper and eliminate fixed delays.
> - **State Logs**:
> - Expand `state-logs.json` schema (e.g., `accountsAssets`,
`assetsMetadata`, `balances`, `nonEvmTransactions`) and adjust internal
account structures.
> - Update `state-logs.spec.ts` to normalize dynamic account IDs before
type-map comparison.
> - **Misc**:
>   - Remove a debug `console.log` in hardware wallets helper.
>   - Fix minor comment typo in snap account transfers test.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
a413e5529b02cb9febaabe57cc79a43c18407b87. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

davibroc commented on test/e2e/page-objects/pages/home/bitcoin-homepage.ts:
```diff
@@ -49,6 +49,23 @@ class BitcoinHomepage extends HomePage {
+    }
+    console.log('Send button is enabled');
+    return true;
+  }
```
I will remove

davibroc committed:
Fix review comments

davibroc commented on test/e2e/tests/solana/check-balance.spec.ts:
```diff
@@ -54,7 +54,9 @@ async function mockPrices(mockServer: Mockttp) {
-describe('Check balance', function (this: Suite) {
+// These tests fails with BIP44
+// eslint-disable-next-line mocha/no-skipped-tests
+describe.skip('Check balance', function (this: Suite) {
```
The mocks for the balance on Solana are working now but we will address this after this PR is merged

davibroc commented on test/e2e/tests/transaction/send-eth.spec.js:
```diff
@@ -157,6 +157,8 @@ describe('Send ETH', function () {
             contractAddress,
           );
 
+          await driver.delay(1000);
```
Removed

davibroc commented on test/e2e/snaps/test-snap-namelookup.spec.ts:
```diff
@@ -19,6 +19,7 @@ describe('Name lookup', function () {
         dappOptions: {
           customDappPaths: [DAPP_PATH.TEST_SNAPS],
         },
+        forceBip44Version: false,
```
@FrederikBolding by the default this variable is set to `true` in the `defaultFixture` function so that it mocks the FF response to have the BIP44 functionality. But here is not needed because BIP44 is already available. Also the mock  would overwrite the FF mocked present in this test that turns on `Send Redesign`.

davibroc committed:
Fix tests

davibroc committed:
Fix folders

davibroc committed:
Fix HW wallet

davibroc committed:
Fix more tests

davibroc committed:
Fix token tests

davibroc committed:
Fix token list test

davibroc committed:
Fix failures

davibroc committed:
Fix metric errors

davibroc committed:
Fixed account transfer

davibroc committed:
Fix snap test

davibroc committed:
Add bug information

ccharly commented on test/e2e/tests/multichain-accounts/multichain-account-list-menu.spec.ts:
```diff
@@ -1,61 +1,83 @@
-        state: 2,
+        testSpecificMock: async (mockServer: Mockttp) => {
+          await mockSnapSimpleKeyringAndSite(mockServer);
+          return [await mockPriceApi(mockServer)];
```
Any reason why we're putting this in an array? 🤔 The other usage of this function usually just return `return await mock...` or they don't event return anything.

ccharly commented on test/e2e/tests/multichain-accounts/multichain-wallet-details.spec.ts:
```diff
@@ -1,83 +1,48 @@
-  it('should add new Ethereum account from wallet details', async function () {
-    await withSolanaAccountSnap(
+describe('Multichain Accounts - Wallet Details', function (this: Suite) {
+  it('should view wallet details with one Ethereum', async function () {
```
This test is now completely different and it does not test the wallet details at all 😅

davibroc committed:
Fix lint

davibroc committed:
Fix conflicts

davibroc committed:
Fix flask tests

davibroc committed:
Fix API specs tests

davibroc committed:
Fix flakiness

davibroc committed:
Fix  dapp multichain

davibroc committed:
Fix test dapp

davibroc committed:
Skipped failing test and refactored profile metrics test

davibroc committed:
Fix state log test

ccharly commented on test/e2e/tests/multichain-accounts/common.ts:
```diff
@@ -28,66 +29,65 @@ export async function withMultichainAccountsDesignEnabled(
+        const additionalMocks = testSpecificMock
+          ? await testSpecificMock(mockServer)
+          : [];
+        return [await mockPriceApi(mockServer), [additionalMocks]];
```
To be reworked in a follow-up PR: Not sure why we nest arrays here?

ccharly commented on test/e2e/tests/multichain-accounts/add-account.spec.ts:
```diff
@@ -27,16 +28,17 @@ const importedAccount = {
-  it('should not affect public address when using secret recovery phrase to recover account with non-zero balance', async function () {
+  // BUG #38568 - Sending token crashes the Extension with BigNumber error
+  // eslint-disable-next-line mocha/no-skipped-tests
+  it.skip('should not affect public address when using secret recovery phrase to recover account with non-zero balance', async function () {
```
To be reworked in a follow-up PR: Double-check if this is now fixed

ccharly commented on test/e2e/tests/identity/account-syncing/multi-srp.spec.ts:
```diff
@@ -107,22 +105,14 @@ describe('Account syncing - Multiple SRPs', function () {
         await driver.delay(10000);
 
-        // Wait for the import to complete and sync
-        await waitUntilSyncedAccountsNumberEquals(3);
```
To be reworked in a follow-up PR: We should probably re-enable this, and if this is not working anymore, then we have to re-adapt the values most probably, but this seems important for B&S coverage!

ccharly commented on test/e2e/tests/identity/account-syncing/multi-srp.spec.ts:
```diff
@@ -142,7 +130,6 @@ describe('Account syncing - Multiple SRPs', function () {
           SRP_2_SECOND_ACCOUNT,
         );
 
-        await waitUntilSyncedAccountsNumberEquals(4);
```
To be reworked in a follow-up PR: Same here

ccharly commented on test/e2e/tests/account/create-remove-account-snap.spec.ts:
```diff
@@ -33,7 +33,8 @@ describe('Create and remove Snap Account', function (this: Suite) {
         const headerNavbar = new HeaderNavbar(driver);
-        await headerNavbar.checkAccountLabel('SSK Account');
+        // BUG #37591 - With BIP44 the account mame is not retained.
+        await headerNavbar.checkAccountLabel('Snap Account 1');
```
To be reworked in a follow-up PR: This is true, and IDK if we should consider this a bug or not (since we're doing a some rework in the Snap account flows). To be checked!

> [!NOTE]
> We should check other mentions of this bug in all E2E too!

ccharly commented on test/e2e/tests/account/snap-account-transfers.spec.ts:
```diff
@@ -75,25 +72,33 @@ describe('Snap Account Transfers', function (this: Suite) {
-        await accountList.checkAccountBalanceDisplayed('$44,200');
-        await accountList.checkAccountBalanceDisplayed('$40,799');
+        // Account balance doesn't update after transaction is completed
+        // await accountList.checkMultichainAccountBalanceDisplayed('$88,426');
```
To be reworked in a follow-up PR: Maybe rework the test to force-load the balances (by clicking on each accounts)

ccharly commented on test/e2e/tests/account/snap-account-transfers.spec.ts:
```diff
@@ -125,17 +133,20 @@ describe('Snap Account Transfers', function (this: Suite) {
-        await accountList.checkAccountBalanceDisplayed('$44,200');
-        await accountList.checkAccountBalanceDisplayed('$40,799');
+        // Account balance doesn't update after transaction is completed
+        // await accountList.checkMultichainAccountBalanceDisplayed('$88,426');
```
To be reworked in a follow-up PR: Maybe rework the test to force-load the balances (by clicking on each accounts)

ccharly commented on test/e2e/tests/hardware-wallets/qr-account.spec.ts:
```diff
@@ -91,22 +90,20 @@ describe('QR Hardware', function () {
-        await homePage.checkExpectedBalanceIsDisplayed('0');
-        await headerNavbar.openAccountMenu();
-        await accountListPage.checkAccountIsNotDisplayedInAccountList('QR 1');
+        // Remove Ledger 1 account and check Ledger 1 account is removed
```
To be reworked in a follow-up PR: Wrong comment here, Ledger -> QR

ccharly commented on test/e2e/tests/onboarding/onboarding.spec.ts:
```diff
@@ -300,7 +317,9 @@ describe('MetaMask onboarding', function () {
-  it('Provides an onboarding path for a user who has restored their account from state persistence failure', async function () {
+  // But #38077 - After estoring account from state persistence failure Metamask unlock is not working
+  // eslint-disable-next-line mocha/no-skipped-tests
+  it.skip('Provides an onboarding path for a user who has restored their account from state persistence failure', async function () {
```
To be reworked in a follow-up PR:
- But -> Bug
- Investigate if the accounts-framework states are valid during this scenario?
- Try to re-enable this test

ccharly commented on test/e2e/helpers.js:
```diff
@@ -174,7 +173,7 @@ async function withFixtures(options, testSuite) {
     manifestFlags,
     solanaWebSocketSpecificMocks = [],
-    forceBip44Version = 0,
+    forceBip44Version = true,
```
To be reworked in a follow-up PR: To clean up after state 2 is ON everywhere

ccharly reviewed (dismissed):
LGTM for accounts. I did add a lot of discussions mainly for my team to review later on and rework those areas, so please, let them open for now! (I did unresolved some Cursor discussions too, as they seem legit to me in our context)

Gudahtt reviewed (dismissed):
Approving for privacy reviewers team, state logs look OK 👍
